Eevee.Sleep.Bot/Workers/InGameAnnouncementCrawlingWorker.cs

```diff
@@ -0,0 +1,93 @@
+    private int retryCount = 0;
+
+    private static readonly TimeSpan CheckInterval = TimeSpan.FromSeconds(60);
+
+    private static async Task<IEnumerable<InGameAnnouncementIndexModel>> GetIndexes() {
+        Dictionary<string, InGameAnnoucementLanguage> BaseUrls = new(){
+            { "https://www.pokemonsleep.net/news", InGameAnnoucementLanguage.JP },
+            { "https://www.pokemonsleep.net/en/news", InGameAnnoucementLanguage.EN },
+            { "https://www.pokemonsleep.net/zh/news", InGameAnnoucementLanguage.ZH },
+        };
+
+        var tasks = BaseUrls.Select(dict => IndexScraper.GetAllPagesAsync(dict.Key, dict.Value)).ToArray();
+        var results = await Task.WhenAll(tasks);
+
+        return results.SelectMany(x => x);
+    }
+
+    private static async Task<IEnumerable<InGameAnnouncementDetailModel>> GetDetails(IEnumerable<InGameAnnouncementIndexModel> indexes) {
+        var detailTasks = indexes
+            .AsParallel()
+            .WithDegreeOfParallelism(5)
+            .Select(DetailScraper.GetAsync);
+    
+        return await Task.WhenAll(detailTasks);
+    }
+
+    private static async Task SaveDetailsAndHistories(IEnumerable<InGameAnnouncementDetailModel> details) {
+        var existedDetails = InGameAnnouncementDetailController.FindAllByIds(details.Select(x => x.AnnouncementId));
+        var existedDetailsById = existedDetails.ToDictionary(x => x.AnnouncementId);
+    
+        var shouldSaveDetail = new List<InGameAnnouncementDetailModel>();
+    
+        foreach (var detail in details) {
+            var detailModel = existedDetailsById.GetOrDefault(detail.AnnouncementId, null);
+            if (detailModel?.ContentHash != detail.ContentHash) {
+                shouldSaveDetail.Add(detail);
+            }
+        }
+        await InGameAnnouncementDetailController.BulkUpsert([..shouldSaveDetail]);
+        await InGameAnnouncememntHistoryController.BulkUpsert([..shouldSaveDetail]);
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken) {
+        _logger.LogInformation("Starting in-game announcement update watcher.");
+        cancellationToken.Register(() => _logger.LogInformation("Stopping in-game announcement update watcher: cancellation token received."));
+
+        while (!cancellationToken.IsCancellationRequested) {
+            _logger.LogInformation("Checking in-game announcement updates.");
+
+            try {
+                var indexes = await GetIndexes();
+                await InGameAnnouncememntIndexController.BulkUpsert([..indexes]);
+                
+                var details = await GetDetails(indexes);
+                await SaveDetailsAndHistories(details);
+
+                retryCount = 0;
+            } catch (ContentStructureChangedException e) {
+                _logger.LogError("Failed to get indexes. Web page structure may have changed. retries: {RetryCount}", retryCount);
+                if(++retryCount >= MAX_RETRY_COUNT) {
```

Comment:
Add an extra space after `if (`

Also although I can understand it, put `retryCount++` after the if bracket instead of doing inline expression with assignment `++retryCount` for clarity purpose